FirebaseCore/Tests/SwiftUnit/CoreAPITests.swift

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// MARK: This file is used to evaluate the experience of using Firebase APIs in Swift.
+
+import Foundation
+
+import FirebaseCore
+
+final class CoreAPITests {
+  func usage() {
+    // MARK: - FirebaseApp
+
+    // Configure Firebase app
+    FirebaseApp.configure()
+
+    if let options = FirebaseOptions(contentsOfFile: "path/to/GoogleService-Info.plist") {
+      FirebaseApp.configure(name: "App", options: options)
+      FirebaseApp.configure(options: options)
+    }
+
+    // Retrieve Firebase app(s)
+    if let _ /* app */ = FirebaseApp.app() {
+      // ...
+    }
+
+    if let _ /* app */ = FirebaseApp.app(name: "App") {
+      // ...
+    }
+
+    if let _ /* apps */ = FirebaseApp.allApps {
+      // ...
+    }
+
+    // Delete Firebase app
+    if let app = FirebaseApp.app() {
+      app.delete { _ /* succes */ in
+        // ...
+      }
+    }
+
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            try await app.delete()
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+
+    // Properties
+    if let app = FirebaseApp.app() {
+      _ = app.name
+      _ = app.options
+      _ = app.isDataCollectionDefaultEnabled
+      app.isDataCollectionDefaultEnabled = false
+    }
+
+    // MARK: - FirebaseConfiguration
+
+    _ = FirebaseConfiguration.shared
+    FirebaseConfiguration.shared.setLoggerLevel(.debug)
+
+    // MARK: - FirebaseLoggerLevel
+
+    let loggerLevel: FirebaseLoggerLevel = .debug
+    switch loggerLevel {
+    case FirebaseLoggerLevel.error:
+      break
+    case FirebaseLoggerLevel.warning:
+      break
+    case FirebaseLoggerLevel.notice:
+      break
+    case FirebaseLoggerLevel.info:
+      break
+    case FirebaseLoggerLevel.debug:
+      break
+    default:
+      break
+    }
+
+    _ = FirebaseLoggerLevel.min
```

Comment:
Is it Swift version dependent whether these can be in the switch?

Reply:
I don't believe so. Including them in the switch led to a [warning which failed CI](https://github.com/firebase/firebase-ios-sdk/runs/3015446535?check_suite_focus=true#step:4:32). The warning was that the `min` and `max` case would never be executed. I think it's just because `min` and `max` point to enum cases that the switch would always catch first.